test/parallel_api/iterator/iterators.pass.cpp

```diff
@@ -162,10 +162,10 @@ void test_explicit_move(InputIterator i, InputIterator j) {
 struct test_zip_iterator {
     template <typename T1, typename T2>
     void operator()(::std::vector<T1>& in1, ::std::vector<T2>& in2) {
-        //test that zip_iterator is default constructible
-        oneapi::dpl::zip_iterator<decltype(in1.begin()), decltype(in2.begin())> b;
+        static_assert(::std::is_default_constructible<oneapi::dpl::zip_iterator<decltype(in1.begin()), decltype(in2.begin())>>::value,
+        "Zip_iterator is not default constructible");
 
-        b = oneapi::dpl::make_zip_iterator(in1.begin(), in2.begin());
+        auto b = oneapi::dpl::make_zip_iterator(in1.begin(), in2.begin());
```

Comment:
Removing test for `operator=` we reduce the test coverage

Reply:
Resolved it with new static_assert

Reply:
I don't believe that compile time check increases the coverage.

Another question. Why logical test is mixed with coverage test? If we need something for coverage it should be either separate or well-commented. Otherwise, the previously existing lines don't make sense without context

Reply:
Decided to revert changes and clarified comments